tests/scripts/analyze_outcomes.py

```diff
@@ -22,17 +22,20 @@ def __init__(self):
         self.error_count = 0
         self.warning_count = 0
 
-    @staticmethod
-    def log(fmt, *args, **kwargs):
-        sys.stderr.write((fmt + '\n').format(*args, **kwargs))
+    def info(self, fmt, *args, **kwargs):
+        self.print_line('Info: ' + fmt, *args, **kwargs)
 
     def error(self, fmt, *args, **kwargs):
-        self.log('Error: ' + fmt, *args, **kwargs)
         self.error_count += 1
+        self.print_line('Error: ' + fmt, *args, **kwargs)
 
     def warning(self, fmt, *args, **kwargs):
-        self.log('Warning: ' + fmt, *args, **kwargs)
         self.warning_count += 1
+        self.print_line('Warning: ' + fmt, *args, **kwargs)
+
+    @staticmethod
+    def print_line(fmt, *args, **kwargs):
```

Comment:
Question: does this method need to be static? I thought we wanted to discourage printing stuff without an instance. Also, perhaps we should make it "private" (that is, name it `_print_line`) to discourage calling it directly as opposed to calling the info/warning/error methods above.

Reply:
Ah OK, makes sense!